src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1327,13 +1349,14 @@ impl Store {
     digest: DirectoryDigest,
   ) -> Result<Vec<FileContent>, StoreError> {
     let mut files = Vec::new();
-    self
-      .load_digest_trie(digest)
-      .await?
-      .walk(&mut |path, entry| match entry {
+    self.load_digest_trie(digest).await?.walk(
+      SymlinkBehavior::Aware,
```

Comment:
Is `DigestContents` supposed to be symlink aware or symlink oblivious? I think that a consumer would want to pretend that symlinks do not exist, and get the content of files multiple times in case of symlinks, at least in part because that is symmetrical with `Snapshot`. So this should probably be oblivious.